docs/editor-neovim.qmd

```diff
@@ -0,0 +1,63 @@
+---
+title: "Neovim"
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+# Installing the air client
+
+First, install the Air client by following the instructions on the [Air CLI documentation](https://posit-dev.github.io/air/cli.html).
+
+# Configuration
+
+Air can be configured as a Language Server Protocol (LSP) client via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) or as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim).
+
+## LSP Configuration with nvim-lspconfig
+
+To configure Air as an LSP client, add the following to your `nvim/lua/plugins/lspconfig.lua`:
+
+```lua
+require("lspconfig").air.setup({})
+```
+
+To enable automatic formatting on save, add the following to your `nvim/after/plugin/lsp.lua` (or other appropriate location):
+
+```lua
+vim.api.nvim_create_autocmd("BufWritePre", {
```

Comment:
Do you think it makes sense to provide this autocmd? I'd expect everyone to already have something like this in their config.

However, for the completeness (e.g. for Neovim newcomers) it might make sense to keep it.

Reply:
Worth to consider: using `ftplugin`.

Putting the below command in either `ftplugin/r.lua` or `after/ftplugin/r.lua` would have a roughly the same effect. I supposed it's a style preference of using `pattern` vs. filetypes.
```lua
vim.api.nvim_create_autocmd("BufWritePre", {
  buffer = 0,
  callback = function()
    vim.lsp.buf.format()
  end,
})
```

Reply:
> Worth to consider: using `ftplugin`.
> 
> Putting the below command in either `ftplugin/r.lua` or `after/ftplugin/r.lua` would have a roughly the same effect. I supposed it's a style preference of using `pattern` vs. filetypes.
> 
> ```lua
> vim.api.nvim_create_autocmd("BufWritePre", {
>   buffer = 0,
>   callback = function()
>     vim.lsp.buf.format()
>   end,
> })
> ```

I think you are right, either are fine. Maybe we can propose both solution/approach.

Reply:
When responding to another comment, I just realized that `on_attach` seems to make the all-in-one `setup()` call:
```lua
require("lspconfig").air.setup({
  on_attach = function(_, bufnr)
    vim.api.nvim_create_autocmd("BufWritePre", {
      buffer = bufnr,
      callback = function()
        vim.lsp.buf.format()
      end,
    })
  end,
})
```

Reply:
I will use your solution, it makes sens and likely easier to configure for neovim newcomer. Thank you.